File: backend/app/controllers/health.py
from fastapi import APIRouter

from app.config import settings

router = APIRouter(tags=["health"])


@router.get("/health")
def get_health() -> dict[str, str]:
    """Liveness check used by the Docker healthcheck and the frontend's
    update-check banner (see frontend's `api/health.ts`)."""
    return {"status": "ok", "version": settings.app_version}

File: frontend/src/api/health.ts
export interface HealthStatus {
  status: string;
  version: string;
}

export async function getHealth(): Promise<HealthStatus> {
  const response = await fetch("/api/health");
  if (!response.ok) {
    throw new Error(`Health check failed: ${response.status}`);
  }
  return (await response.json()) as HealthStatus;
}
